src/test/java/uk/gov/companieshouse/api/accounts/transformer/StocksTransformerTest.java

```diff
@@ -0,0 +1,224 @@
+
+        stocksEntity.setData(stocksDataEntity);
+
+        Stocks stocks = stocksTransformer.transform(stocksEntity);
+
+        assertNotNull(stocks);
+        assertEqualsRestObject(stocks);
+        assertEquals(new HashMap<>(), stocks.getLinks());
+        assertEquals(ETAG, stocks.getEtag());
+        assertEquals(KIND, stocks.getKind());
+    }
+
+    @Test
+    @DisplayName("Tests transformer with fully populated Entity object and validates values returned")
+    public void testEntityToRestTransformerWithFullyPopulatedEntityObject() {
+
+        StocksEntity stocksEntity = new StocksEntity();
+        StocksDataEntity stocksDataEntity = new StocksDataEntity();
+
+        stocksDataEntity.setEtag(ETAG);
+        stocksDataEntity.setKind(KIND);
+        stocksDataEntity.setLinks(new HashMap<>());
+        stocksDataEntity.setCurrentPeriodEntity(createCurrentPeriodEntityObject());
+        stocksDataEntity.setPreviousPeriodEntity(createPreviousPeriodEntityObject());
+
+        stocksEntity.setData(stocksDataEntity);
+
+        Stocks stocks = stocksTransformer.transform(stocksEntity);
+
+        assertNotNull(stocks);
+        assertEqualsRestObject(stocks);
+        assertEquals(new HashMap<>(), stocks.getLinks());
+        assertEquals(ETAG, stocks.getEtag());
+        assertEquals(KIND, stocks.getKind());
+    }
+
+    private PreviousPeriodEntity createPreviousPeriodEntityObject() {
+
+        PreviousPeriodEntity previousPeriodEntity = new PreviousPeriodEntity();
+        previousPeriodEntity.setPaymentsOnAccount(PAYMENTS_ON_ACCOUNT_PREVIOUS_PERIOD);
+        previousPeriodEntity.setStocks(STOCKS_PREVIOUS_PERIOD);
+        previousPeriodEntity.setTotal(TOTAL_PREVIOUS_PERIOD);
+
+        return previousPeriodEntity;
+    }
+
+    private CurrentPeriodEntity createCurrentPeriodEntityObject() {
+
+        CurrentPeriodEntity currentPeriodEntity = new CurrentPeriodEntity();
+        currentPeriodEntity.setPaymentsOnAccount(PAYMENTS_ON_ACCOUNT_CURRENT_PERIOD);
+        currentPeriodEntity.setStocks(STOCKS_CURRENT_PERIOD);
+        currentPeriodEntity.setTotal(TOTAL_CURRENT_PERIOD);
+
+        return currentPeriodEntity;
+    }
+
+    private PreviousPeriod createPreviousPeriodRestObject() {
+
+        PreviousPeriod previousPeriod = new PreviousPeriod();
+        previousPeriod.setPaymentsOnAccount(PAYMENTS_ON_ACCOUNT_PREVIOUS_PERIOD);
+        previousPeriod.setStocks(STOCKS_PREVIOUS_PERIOD);
+        previousPeriod.setTotal(TOTAL_PREVIOUS_PERIOD);
+
+        return previousPeriod;
+    }
+
+    private CurrentPeriod createCurrentPeriodRestObject() {
+
+        CurrentPeriod currentPeriod = new CurrentPeriod();
+        currentPeriod.setPaymentsOnAccount(PAYMENTS_ON_ACCOUNT_CURRENT_PERIOD);
+        currentPeriod.setStocks(STOCKS_CURRENT_PERIOD);
+        currentPeriod.setTotal(TOTAL_CURRENT_PERIOD);
+
+        return currentPeriod;
+    }
+
+    private void assertEqualsEntityObject(StocksEntity stocksEntity) {
```

Comment:
we could call this something like `validateReturnedStockEntityObject()` instead of `assertEquals...`

Reply:
that's perfect. I like it more thant `assert....` ;)